### DISCUSSION TITLE
[Request for Feedback] Node Operators the Ecotone Network Upgrade requires a CL Client

### DISCUSSION THREAD

**Discussion by sbvegan** [ethereum-optimism/developers#234] [Feedback]
## Request for Feedback

Hey y'all,

With [EIP-4844 Blobs](https://www.eip4844.com/) around the corner, the next proposed network upgrade, Ecotone will require the `op-node` to configure a Beacon endpoint. This means node operators will need access to a [consensus client](https://ethereum.org/developers/docs/nodes-and-clients#consensus-clients) (i.e. Lighthouse, Lodestar, Nimbus, Prysm, Teku, etc.).

I'd like to know how this effects you as a node operator. Are you using a RPC provider for your Ethereum L1 node? Are you running your own Ethereum L1 node and this is no problem? Please share your thoughts in this thread.

Kind regards,
Soyboy
Developer Support Engineer - OP Labs

## Additional Information

It would be helpful to know:

- know about your node setup and what you use your node for
- the questions you have around Ecotone and a post-blob world

### Documentation 

[Ecotone Network Upgrade Documentation](https://docs.optimism.io/builders/node-operators/network-upgrades/ecotone)
[Software Release Documentation](https://docs.optimism.io/builders/node-operators/releases)

**Comment by ghost** [comment]
I have asked a question about the recommended L1 CL node configuration since they prune historic blobs and I wonder if op-node could run on the top of pruned blobs https://github.com/ethereum-optimism/developers/discussions/236

**Comment by Chomtana** [comment]
Most node operators are using an external RPC provider for the Ethereum L1 node. Wondering if we can run consensus client (i.e. Lighthouse, Lodestar, Nimbus, Prysm, Teku, etc.) independently from ETH full node.

If consensus client is independent from ETH full node then it shouldn't be a problem. I think each RPC provider is going to provide public beacon chain access soon.

Also, if custom OP Stack chain use external DA like Celestia or not opt into EIP-4844, would these chain need consensus client?

**Reply by sbvegan** [reply]
> Wondering if we can run consensus client (i.e. Lighthouse, Lodestar, Nimbus, Prysm, Teku, etc.) independently from ETH full node.

I know that the EL and CL nodes are generally run in a 1:1 setup, so I'm not sure about this, I'll have to ask someone else.

> I think each RPC provider is going to provide public beacon chain access soon.

As of now, Quicknode provides a beacon endpoint, but others do not. Alchemy is considering opening one up if there's customer demand. Infura used to have it, but there wasn't demand for it so they stopped providing it as a service.

> Also, if custom OP Stack chain use external DA like Celestia or not opt into EIP-4844, would these chain need consensus client?

The primary use for this endpoint is for blob data posted to L1 so users can get cheaper fees by reducing the calldata costs. I don't know the numbers but I suspect alt-DA can still be cheaper. 

This functionality is built into the node software, so if you're running the latest support release with it you'll need a consensus client. Folks who have hacked their node will need to maintain a fork and take in upstream changes as new releases of the software comes out if they want to avoid using the CL. It makes sense to keep their forks as close to the upstream code as possible to reduce the maintenance burden, so they might still include the functionality. At the end of the day, its up to the software maintainer to decide. Did you have a specific chain in mind when you asked about this?

**Comment by AmardeepSambaru** [comment]
Hi Good morning.

I updated the binary of op-node and op-geth in optimism Sepolia and started the node with --l1.beacon and i have beacon endpoint.
But is given the below error. and also iam facing same issue in Base Sepolia node it is also given same error.

Can you Please Help me.
Thank you

t=2024-02-20T05:31:23+0000 lvl=eror msg="Error initializing the rollup node" err="failed to check L1 Beacon API version: operation failed permanently after 5 attempts: invalid character 'N' looking for beginning of value"
t=2024-02-20T05:31:23+0000 lvl=crit msg="Application failed" message="failed to setup: unable to create the rollup node: failed to check L1 Beacon API version: operation failed permanently after 5 attempts: invalid character 'N' looking for beginning of value"

**Reply by Chomtana** [reply]
Invalid character N may mean beacon endpoint response with "Not Found"

**Comment by opfocus** [comment]
Here's a successful configuration shared by a developer:
For Prysm configuration:
`--grpc-gateway-host=0.0.0.0 --grpc-gateway-port=XXXX`
For OP node configuration:
`--l1.beacon=http://IP:XXXX`
Source: https://discord.com/channels/667044843901681675/1080862392281481246/1209857519053312010
